### PR TITLE
Restore non-Google search

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #14606)

### CODE REVIEW

**Comment by aeslaughter** [comment]
@sapitts Can you try this out with BISON to make sure it is working

**Comment by sapitts** [comment]
Grabbing your branch now as the submodule in the tagged version of BISON that I have. Thanks for working on this for us!

**Comment by sapitts** [comment]
This successfully restores the search functionality for the tagged version of BISON; tested via a local build with this branch as the MOOSE submodule

**Comment by sapitts** [comment]
fyi @novasr since we talked about this earlier today

File: python/MooseDocs/extensions/navigation.py [hunk 1]
```diff
@@ -120,6 +126,38 @@ def postRender(self, page, result):
...
+
+            id_ = head.get('id')
+            if id_ == '':
+                id_ = head.text('-').lower()
+            index.append(dict(title=title, text=head.text(), bookmark=id_))
+        page['search'].extend(index)
+
+
+    def postExecute(self):
+        """Build the JSON file containing the index data."""
```
Comment (sapitts) [inline python/MooseDocs/extensions/navigation.py]: Would it be worthwhile to expand this comment to include some information about the use cases for the apps that don't use the google search functionality?

**Review by sapitts — DISMISSED** [review]

**Review by permcody — APPROVED** [review]

**Pull request merged**